### DISCUSSION TITLE
setPage method adds unexpected properties (v2.x)

### DISCUSSION THREAD

**Discussion by livelogic** [livewire/livewire#5937] [7. Bugs (archived)] [answered]
In v2.x, the method `setPage($page, $pageName = 'page')` in the `WithPagination` trait sets (or overwrites) properties named like a paginator. This may cause conflicts in the component namespace.
Lines 107-109:

```
$this->paginators[$pageName] =  $page;

$this->{$pageName} = $page;
```

I notice this has been fixed in the main repository, where the logic has been moved to a new trait `HandlesPagination`, which doesn't have the issue.
Line 61:
```
$this->paginators[$pageName] = $page;
```

Can this be fixed in v2.x as well?

**Comment by joshhanley** [comment, marked as answer]
@livelogic in V2, the names of the paginators were set as public properties. The `$this->paginators[]` property came later when we supported multiple paginators on a page. But sadly, we can't remove the old format as that would be a breaking change for those that reply on those properties existing.

I would recommend just ensuring that you don't use any public properties with the same name as your paginators.

Hope this helps!
